src/controllers/mainController.ts

```diff
@@ -262,6 +273,11 @@ export default class MainController implements vscode.Disposable {
 					conn.password = '';
 					profileChanged = true;
 				}
+				// Fixup 'Encrypt' property if needed
+				let result = ConnInfo.updateEncrypt(conn);
+				if (result.updateStatus) {
+					await this.connectionManager.connectionStore.saveProfile(result.connection as IConnectionProfile);
```

Comment:
Do you need to call saveProfile here vs just setting profileChanged to true? Maybe even call saveProfile instead of setConfiguration directly since that seems like what we want to be doing anyways...

Reply:
SaveProfile is needed here, just for the same reason as 'saveProfilePasswordIfNeeded' is called above.

Reply:
I was more asking why we have different ways of "saving" a profile in this function. It just makes it confusing to try and follow and can easily lead to errors down the line if someone doesn't realize how these two differ.

And I would more think that we should be calling saveProfile in all cases when a profile is changed. I'd have to spend more time following the code path to understand why we'd want to save the profile in one case but not another.

If you've done that yourself and it makes sense then that's fine (although i would love to see some comments added to help future people looking at this...).